web/src/copy_and_paste.ts

```diff
@@ -580,7 +632,7 @@ export function paste_handler(event) {
         return;
     }
 
-    if (clipboardData.getData) {
+    if (clipboardData.getData && event.currentTarget instanceof HTMLTextAreaElement) {
```

Comment:
Are we expecting `event.currentTarget` be of a type other than `HTMLTextAreaElement`? If not than an `assert` might be better.